### PR TITLE
Create Participants Field in Project Settings page 

### LINKED ISSUE #2030: Create Participants Field in Project Settings page
Is a subtask for #1810 
image

### PR DESCRIPTION
**Pull request by asabotovich**
## PR includes

- [x] Feature

## Related issues

Resolve #2030 

## QA Instructions, Screenshots, Recordings
  
image

### CODE REVIEW

File: src/components/ProjectSettingsPage/CustomFormField.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    background-color: transparent;
+`;
+
+const StyledChildrenContainer = styled.div`
+    flex: 1;
+    padding: ${gapXs} ${gapM};
+`;
+
+export const CustomFormField: FC<{ children: ReactNode; label: string; name?: string }> = ({
```
Comment (asabotovich) [inline src/components/ProjectSettingsPage/CustomFormField.tsx:32]: This should be moved to bricks after review

**Comment by asabotovich** [comment]
https://github.com/taskany-inc/issues/assets/9020793/d58ca7f8-740e-4b67-8f23-9bfe6ea34989

File: src/components/ProjectPageTabs/ProjectPageTabs.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import { useRouter } from 'next/router';
+import { TabsMenu, TabsMenuItem, nullable } from '@taskany/bricks';
+
+import { routes } from '../../hooks/router';
+import { pageTabs, pageActiveTabItem } from '../../utils/domObjects';
+
+import { tr } from './ProjectPageTabs.i18n';
+
+export const ProjectPageTabs: FC<{ id: string; editable?: boolean }> = ({ id, editable = false }) => {
+    const tabsMenuOptions: Array<[string, string, boolean]> = useMemo(
```
Comment (tsumo) [inline src/components/ProjectPageTabs/ProjectPageTabs.tsx:12]: suggestion: `Array<[title: string, href: string, ownerOnly: boolean]>`

File: trpc/router/project.ts [hunk 3]
```diff
@@ -522,18 +522,35 @@ export const project = router({
             const parentsToConnect = parent?.filter((pr) => !project.parent.some((p) => p.id === pr.id));
             const parentsToDisconnect = project.parent.filter((p) => !parent?.some((pr) => p.id === pr.id));
 
+            const participantsToConnet = participants?.filter(
```
Comment (tsumo) [inline trpc/router/project.ts:525]: typo - ToConnet

File: trpc/router/project.ts [hunk 4]
```diff
@@ -522,18 +522,35 @@ export const project = router({
...
+                (pr) => !project.participants.some((p) => p.id === pr.id),
+            );
+            const participantsToDisconnect = project.participants.filter(
```
Comment (tsumo) [inline trpc/router/project.ts:528]: Does this mean that you have to always pass all current participants when you want to update project?

Comment (asabotovich) [reply trpc/router/project.ts:528]: Yes, we have some logic for parents several line above and for goal tag

**Review by tsumo — APPROVED** [review]

File: src/components/AddInlineTrigger.tsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    height: 28px; // Input height
+`;
+
+interface AddInlineTriggerProps {
+    text: string;
+    onClick: ComponentProps<typeof InlineTrigger>['onClick'];
+    icon?: React.ReactNode;
+}
+
+export const AddInlineTrigger = forwardRef<HTMLDivElement, AddInlineTriggerProps>(
```
Comment (LamaEats) [inline src/components/AddInlineTrigger.tsx:18]: But why? Why you don't using default `InlineTrigger` component?

Comment (asabotovich) [reply src/components/AddInlineTrigger.tsx]: became a part of UserEditableList

**Review by LamaEats — APPROVED** [review]

File: src/components/ProjectParticipants/ProjectParticipants.tsx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+            updateProject()({
+                ...project,
+                participants: [...project.participants, user],
+            });
+        },
+        [updateProject, project],
+    );
+
+    useEffect(() => {
+        if (userToRemoveGoals && userToRemoveGoals.length) {
```
Comment (DenisVorop) [inline src/components/ProjectParticipants/ProjectParticipants.tsx:55]: you can remove existence check `userToRemoveGoals` if set default value `const { data: userToRemoveGoals = [] }`

File: src/components/ProjectParticipants/ProjectParticipants.tsx [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+import { trpc } from '../../utils/trpcClient';
+import { ActivityByIdReturnType, ProjectByIdReturnType } from '../../../trpc/inferredTypes';
+import { useProjectResource } from '../../hooks/useProjectResource';
+import { ModalEvent, dispatchModalEvent } from '../../utils/dispatchModal';
+import { UserEditableList } from '../UserEditableList/UserEditableList';
+import { SettingsCard } from '../SettingsContent';
+
+import { tr } from './ProjectParticipants.i18n';
+
+type User = NonNullable<ActivityByIdReturnType>;
```
[the remark this answers is not in the document]
Comment (asabotovich) [reply src/components/ProjectParticipants/ProjectParticipants.tsx]: fixed

File: src/components/UserEditableList/UserEditableList.tsx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    text: string;
+    onClick: ComponentProps<typeof InlineTrigger>['onClick'];
+    icon?: React.ReactNode;
+}
+
+export const UserEditableListTrigger = forwardRef<HTMLDivElement, UserEditableListTriggerProps>(
+    ({ icon = <IconPlusCircleOutline size="xs" />, text, onClick }, ref) => (
+        <StyledInlineTrigger ref={ref} icon={icon} text={text} onClick={onClick} />
+    ),
+);
```
[the remark this answers is not in the document]
Comment (asabotovich) [reply src/components/UserEditableList/UserEditableList.tsx:38]: fixed

**Pull request merged**